web/src/main/webapp/table.ts

```diff
@@ -765,7 +778,19 @@ export class TableView extends RemoteObjectView
         }
     }
 
+    private heatMapArray(): void {
+        let selectedColumns: string[] = [];
+        this.selectedColumns.forEach((col) => selectedColumns.push(col));
```

Comment:
apparently there are simpler ways to clone an array in javascript.
Maybe you want to add a util.ts function that clones an array?

Reply:
The problem here is that `this.selectedColumns` is a set, not an array. But I added functions for both cases in `util.ts` now, so it's simpler.